scripts/install_prereqs.py

```diff
@@ -0,0 +1,453 @@
+
+    This does the following,
+        - initialize/update the vcpkg git submodule
+        - build vcpkg executable
+        - install packages via vcpkg.
+    Args:
+        arch (str): Architecture (eg: 'x86', 'x64').
+        install_vcpkg (bool): Build/Install vcpkg before building packages?
+    """
+    # vcpkg is a submodule of the main cpp repo
+    # Ensure that the submodule is initialized and updated
+    run_command(['git', 'submodule', 'init'])
+    run_command(['git', 'submodule', 'update'])
+
+    if build_vcpkg:
+        # build and install vcpkg
+        script_absolute_path = get_vcpkg_installation_script_path()
+        run_command([script_absolute_path])
+    else:
+        print("Skipping building/installing vcpkg executable")
+
+    # for each desktop platform, there is an existing vcpkg response file 
+    # in the repo (external/vcpkg_<triplet>_response_file.txt) defined for each target triplet
+    vcpkg_triplet = get_vcpkg_triplet(arch)
+    print('vcpkg triplet auto determined: {0}'.format(vcpkg_triplet))
+    vcpkg_response_file_path = get_vcpkg_response_file_path(vcpkg_triplet)
+    vcpkg_executable_file_path = get_vcpkg_executable_file_path()
+
+    run_command([vcpkg_executable_file_path, 'install', '@' + vcpkg_response_file_path, '--disable-metrics'])
+
+
+def step_install_system_packages(args):
+    """ Install packages (dependencies) with system package manager.
+
+    Args:
+        args (`argparse.Namespace`): Parsed command line arguments.
+    """
+    system_packages_to_install = []
+    # Install ninja and/or ccache if needed
+    # check if ninja is already installed before attempting to install it
+    if args.generator == 'Ninja':
+        if is_tool_installed('ninja'):
+            print('Ninja requested but is already installed. Skipping installation...')
+        elif is_linux_os():
+            system_packages_to_install.append('ninja-build')
+        elif is_mac_os:
+            system_packages_to_install.append('ninja')   
+
+    # ccache is supported only on mac and linux and check if it is already installed
+    if args.ccache and (is_linux_os() | is_mac_os()) and not is_tool_installed('ccache'):
+        if is_tool_installed('ccache'):
+            print('ccache requested but is already installed. Skipping installation...')
+        elif is_linux_os() or is_mac_os():
+            system_packages_to_install.append('ccache')
+
+    if system_packages_to_install:
+        install_system_packages(system_packages_to_install)
+
+
+def step_install_python_packages(args):
+    """ Install python packages (dependencies).
+
+    Args:
+        args (`argparse.Namespace`): Parsed command line arguments.
+    """
+    # Check for packages installed in this python environment
+    installed_packages = {pkg.key for pkg in pkg_resources.working_set}
+
+    # Get the list of packages we intend to install
+    requirements_file_path = get_pip_requirements_file_path()
+    packages_to_install = None
```

Comment:
This initialization is unnecessary, as scope in Python is a bit more permissive. You can initialize a value inside of an intended block (such as 'with', 'if', etc) and reference it outside of the block.

Reply:
Good to know!